Makefile

```diff
@@ -21,9 +21,9 @@ ANDROID_BUILD := $(ANDROID_BUILD_DIR)/Lndmobile.aar
 
 COMMIT := $(shell git describe --tags --dirty)
 
-GOBUILD := go build -v
-GOINSTALL := go install -v
-GOTEST := go test
+GOBUILD := GOEXPERIMENT=loopvar go build -v
+GOINSTALL := GOEXPERIMENT=loopvar go install -v
+GOTEST := GOEXPERIMENT=loopvar go test
```

Comment:
This will break anyone running this with a Go version smaller than `v1.21.0`, so we should probably update the minimum version in `go.mod` and the README/installation instructions.

The alternative would be to add some Makefile magic to detect the Golang version and add this conditionally? Shouldn't be an issue for the `release.sh` script, as we require that to be run with Go `v1.21.0` to produce correct binary hashes.

Reply:
Hmm, yeah good point. Technically, it's enough that we just require this when running tests, as assuming adequate branch coverage, the relevant for loop behaviors across the codebase _should_ be covered. I added it here for build/install as well, as there invariably sections not tested directly rn, and we still want to enforce the correct behavior there. 

I think a middle ground is to: leave it for tests here, but then add a slightly diff makefile target that only CI runs to build+release.

Reply:
We could also just do this:

```
GO_VERSION := $(shell go version | sed -nre 's/^[^0-9]*(([0-9]+\.)*[0-9]+).*/\1/p')
GO_VERSION_MINOR := $(shell echo $(GO_VERSION) | cut -d. -f2)

LOOPVARFIX := 
ifeq ($(shell expr $(GO_VERSION_MINOR) \>= 21), 1)
	LOOPVARFIX := GOEXPERIMENT=loopvar
endif

GOBUILD := $(LOOPVARFIX) go build -v
GOINSTALL := $(LOOPVARFIX) go install -v
GOTEST := $(LOOPVARFIX) go test
```

Since we're using Go 1.21 or later for our release, unit test and integration test builds, we would get all the benefits without breaking things for previous users.